packages/app/src/components/App/index.js

```diff
@@ -24,12 +26,26 @@ const RainbowFirebaseApp = (props) => {
         });
     }, []);
 
+    useEffect(() => {
+        async function initializeApp() {
+            if (initialize) {
+                setLoading(true);
+                await initialize();
+                setLoading(false);
+            }
+            setIsInitializing(false);
+        }
+        initializeApp();
+    }, [initialize]);
```

Comment:
it shouldn't depend on `initialize` prop, the `useEffect` callback function should be invoked once, the first time the component renders